lib/commands/saveScreenshot.js

```diff
@@ -31,26 +36,28 @@ module.exports = function saveScreenshot(fileName) {
     /*!
      * parameter check
      */
-    if (typeof fileName !== 'string') {
+    if (typeof response !== 'function' && typeof response !== 'string') {
         return callback(new ErrorHandler.CommandError('number or type of arguments don\'t agree with saveScreenshot command'));
     }
 
     var self = this,
-        response = {};
+        screenshot = null;
 
     async.waterfall([
-
         function(cb) {
             self.screenshot(cb);
         },
         function(res, cb) {
-            response.screenshot = res;
-            fs.writeFile(fileName, res.value, "base64", callback);
+            screenshot = new Buffer(res.value, 'base64');
+
+            if (typeof response === 'string') {
+                fs.writeFile(response, screenshot, cb);
+            } else {
+                cb();
+            }
         }
     ], function(err) {
-
-        callback(err, null, response);
```

Comment:
we should stay consistent here, meaning that callback should return with error, result and response as parameters here